packages/webapp/src/components/Animals/AddAnimalsFormCard/AddAnimalsFormCard.tsx

```diff
@@ -73,39 +74,48 @@ export default function AddAnimalsFormCard({
   showRemoveButton,
   onRemoveButtonClick,
   isActive,
+  namePrefix = '',
 }: AddAnimalsFormCardProps) {
-  const { control, watch, register } = useFormContext<FormFields>();
+  const { control, watch, register } = useFormContext();
   const { t } = useTranslation();
-  const watchAnimalCount = watch(FIELD_NAMES.COUNT) || 0;
-  const watchAnimalType = watch(FIELD_NAMES.TYPE);
-  const shouldCreateIndividualProfiles = watch(FIELD_NAMES.CREATE_INDIVIDUAL_PROFILES);
+  const watchAnimalCount = watch(`${namePrefix}.${FIELD_NAMES.COUNT}`) || 0;
+  const watchAnimalType = watch(`${namePrefix}.${FIELD_NAMES.TYPE}`);
+  const shouldCreateIndividualProfiles = watch(
+    `${namePrefix}.${FIELD_NAMES.CREATE_INDIVIDUAL_PROFILES}`,
+  );
+
+  const typeKey = watchAnimalType?.value
+    ? parseUniqueAnimalId(watchAnimalType.value, ANIMAL_ID_ENTITY.TYPE)
+    : {};
+
+  const filteredBreeds = breedOptions.filter(
+    ({ custom_type_id, default_type_id }) =>
+      (custom_type_id && custom_type_id === typeKey.custom_type_id) ||
+      (default_type_id && default_type_id === typeKey.default_type_id),
+  );
```

Comment:
I think it would be beneficial to use `watchAnimalType` as is, so I would change the structure of `breedOptions` to something like this:
```
const breedOptions = [
    ...defaultBreeds.map((defaultBreed) => ({
      label: t(`animal:BREED.${defaultBreed.key}`),
      value: generateUniqueAnimalId(defaultBreed),
      type: generateUniqueAnimalId(defaultBreed.default_type_id),
    })),
    ...customBreeds.map((customBreed) => ({
      label: customBreed.breed,
      value: generateUniqueAnimalId(customBreed),
      type: generateUniqueAnimalId(customBreed.default_type_id || customBreed.custom_type_id),
    })),
  ];
```
Then, filter breeds:
```
const filteredBreeds = breedOptions.filter(({ type }) => type === watchAnimalType);
```

I haven't tested it, so something might not make sense, but I think something similar could be done :)

Reply:
I couldn't make it as clean as that because `generateUniqueAnimalId` expects the full type/breed object (since it uses the `key` property)... maybe there were multiple ways but I just overloaded that function to also take an id (somewhat like) what's shown above... it would still need the prefix though so... again not that clean 😂 

But in the end I think it's still an improvement, if you could take a look! 🙏